### PR TITLE
[SDK] Add a config class to store pf configurations

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/_configuration.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import pydash
+
+from promptflow._sdk._utils import dump_yaml
+
+
+class Configuration(object):
+
+    CONFIG_PATH = Path.home() / ".promptflow" / "pf.yaml"
+    COLLECT_TELEMETRY = "cli.collect_telemetry"
+    USER_ID = "cli.user_id"
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_configuration.py:18]: user_session_id?

File: src/promptflow/promptflow/_sdk/_configuration.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    def get_telemetry_consent(self) -> Optional[bool]:
+        """Get the current telemetry consent value. Return None if not configured."""
+        return self._get_config(key=self.COLLECT_TELEMETRY)
+
+    def set_telemetry_consent(self, value):
+        """Set the telemetry consent value and store in local."""
+        self._set_config(key=self.COLLECT_TELEMETRY, value=value)
+
+    def get_or_set_user_id(self):
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_configuration.py:52]: who will set this?

File: src/promptflow/promptflow/_sdk/_configuration.py (hunk 1 above)
Comment (D-W-) [reply src/promptflow/promptflow/_sdk/_configuration.py:18]: Good catch!

File: src/promptflow/promptflow/_sdk/_configuration.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                f.write(dump_yaml(self.config))
+
+    @classmethod
+    def get_instance(cls):
+        return Configuration()
+
+    def _set_config(self, key, value):
+        """Store config in file to avoid concurrent write."""
+        pydash.set_(self.config, key, value)
+        with open(self.CONFIG_PATH, "w") as f:
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_configuration.py:35]: We should be careful, there might be concurrent writters.

Comment (D-W-) [reply src/promptflow/promptflow/_sdk/_configuration.py:39]: Offline synced. Updated `get_instance` to make sure there will be only 1 config instance for each process. And multi processing wirte is a cornet case. We'll leave as it is for now.

**Review by brynn-code — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**